doc/dev/pylint_checking.md

```diff
@@ -0,0 +1,69 @@
+
+In the Azure SDK for Python repository, in addition to the standard pylint library, there are also custom checkers, within the azure-pylint-guidelines-checker package, that help to customize our libraries to the standards described in the [Azure SDK for Python Guidelines](https://guidelinescollab.github.io/azure-sdk/python_introduction.html).
+
+## How to run Pylint?
+
+One way to run pylint is to run at the package level with tox:
+
+    .../azure-sdk-for-python/sdk/eventgrid/azure-eventgrid>tox -e pylint -c ../../../eng/tox/tox.ini
+
+If you don't want to use tox, you can also install and run pylint on its own:
+
+ - If taking this approach, in order to run with the pylintrc formatting and the custom pylint checkers you must also install the custom checkers and `SET` the pylintrc path. 
+
+        pip install pylint
+        pip install azure-pylint-guidelines-checker --index-url="https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-python/pypi/simple/"
+
+        .../azure-sdk-for-python>SET PYLINTRC="./pylintrc"
+        .../azure-sdk-for-python>pylint ./sdk/eventgrid/azure-eventgrid
+
+    Note that you may see different errors if running a different [version of pylint](https://github.com/Azure/azure-sdk-for-python/blob/fdf7c49ea760b1e1698ebbbac48794e8382d8de5/eng/tox/tox.ini#L90) than the one in CI.
+  
+
+# Ignoring Pylint Checkers
+
+- If there is a pylint checker within your SDK that you wish to ignore for that specific scenario (i.e protected-access) you can ignore it with a comment on the offending line.
+
+    `# pylint:disable=protected-access`  
+    
+- (Not Recommended) Another way to disable a checker is by ignoring the entire package in the [pyproject.toml](https://github.com/Azure/azure-sdk-for-python/blob/main/doc/eng_sys_checks.md#the-pyprojecttoml).
+
+
+# Pylint Warnings and Where to Find Them 
+
+Information on the custom pylint checkers resides [here](https://github.com/Azure/azure-sdk-tools/blob/main/tools/pylint-extensions/azure-pylint-guidelines-checker/README.md). [This table](https://github.com/Azure/azure-sdk-tools/blob/main/tools/pylint-extensions/azure-pylint-guidelines-checker/README.md#rules-list), provides information on each custom check and how to resolve them. 
+
+In addition to being a part of the CI, the custom pylint checkers are also integrated into ApiView. If there are unaddressed warnings, they will show as system comments. 
+
+## Next Pylint
+
+There is now a new step on the CI pipeline called `Run Pylint Next`. This is merely a duplicate of the `Run Pylint` step with the exception that `Run Pylint Next` uses a newer version of pylint and a newer version of the custom pylint checkers.
+
+This next-pylint enviornment can also be run locally through tox:
+
+    tox -e next-pylint -c ../../../eng/tox/tox.ini
+
+The errors generated by the `Run Pylint Next` step will not break your weekly test pipelines, but still you should take note and fix them when you have the time. 
```

Comment:
```suggestion
The errors generated by the `Run Pylint Next` step will not break your weekly test pipelines, but you should still fix them when you have the time so that your client library is up to date for the next release. 
```